src/neuromaps_prime/transforms/surface.py

```diff
@@ -53,3 +62,438 @@ def surface_sphere_project_unproject(
+    """
+    if not metric_in.exists():
+        raise FileNotFoundError(f"Input metric file not found: {metric_in}")
+    if not current_sphere.exists():
+        raise FileNotFoundError(f"Current sphere file not found: {current_sphere}")
+    if not new_sphere.exists():
+        raise FileNotFoundError(f"New sphere file not found: {new_sphere}")
+
+    result = workbench.metric_resample(
+        metric_in=metric_in,
+        current_sphere=current_sphere,
+        new_sphere=new_sphere,
+        metric_out=metric_out,
+        method=method,
+        area_surfs=area_surfs,
+    )
+    if not result.metric_out.exists():
+        raise FileNotFoundError(f"Metric out not found: {metric_out}")
+
+    return result
+
+
+def label_resample(
+    label_in: Path,
+    current_sphere: Path,
+    new_sphere: Path,
+    label_out: str,
+    area_surfs: tuple[Path, Path] | None = None,
+    method: str = "ADAP_BARY_AREA",
+) -> workbench.LabelResampleOutputs:
+    """Resample a surface label from one sphere to another.
+
+    Parameters
+    ----------
+    label_in : Path
+        Path to input label file.
+    current_sphere : Path
+        Path to current spherical surface.
+    new_sphere : Path
+        Path to new spherical surface.
+    label_out : str
+        Path to output label file.
+    method : str, optional
+        Resampling method. Default is 'ADAP_BARY_AREA'.
+
+    Returns:
+    -------
+    result : workbench.LabelResampleOutputs
+        Object containing the path to the output label as result.label_out.
+
+    Raises:
+    ------
+    FileNotFoundError
+        If any input file does not exist.
+    """
+    if not label_in.exists():
+        raise FileNotFoundError(f"Input label file not found: {label_in}")
+    if not current_sphere.exists():
+        raise FileNotFoundError(f"Current sphere file not found: {current_sphere}")
+    if not new_sphere.exists():
+        raise FileNotFoundError(f"New sphere file not found: {new_sphere}")
+
+    result = workbench.label_resample(
+        label_in=label_in,
+        current_sphere=current_sphere,
+        new_sphere=new_sphere,
+        label_out=label_out,
+        area_surfs=area_surfs,
+        method=method,
+    )
+    if not result.label_out.exists():
+        raise FileNotFoundError(f"Label out not found: {result.label_out}")
+
+    return result
+
+
+def _get_hop_output_file(
+    output_file_path: str, source: str, next_target: str, density: str, hemisphere: str
+) -> str:
+    """Generate hop output file path based on parameters."""
+    p = Path(output_file_path)
+    parent = p.parent
+    # Remove all suffixes for the base name
+    base = p.stem  # removes only the last suffix
+    # If the file has double suffix (e.g., .func.gii), remove both
+    if p.suffix == ".gii" and p.with_suffix("").suffix:
+        base = p.with_suffix("").stem
+    # Always use .surf.gii for intermediate sphere files
+    suffix = ".surf.gii"
+    return str(
```

Comment:
Just trying to follow this pattern for the file name - could you provide a little more context for the naming here? If this is effectively naming a file for something like `1` -> `2` -> `3`, really just `1` -> `3`, then I think we should still follow the pattern's that are used in the prep repo. I am just way of having a long file name that may be difficult to interpret.